source/simulator/assembly.cc

```diff
@@ -1701,6 +1807,839 @@ namespace aspect
+
+                    //subface->face
+                    scratch.subface_finite_element_values[introspection.extractors.velocities].get_function_values(current_linearization_point,
+                        scratch.face_current_velocity_values);
+
+                    //get the mesh velocity, as we need to subtract it off of the advection systems
+                    if (parameters.free_surface_enabled)
+                      scratch.subface_finite_element_values[introspection.extractors.velocities].get_function_values(free_surface->mesh_velocity,
+                          scratch.face_mesh_velocity_values);
+
+                    compute_material_model_input_values (current_linearization_point,
+                                                         scratch.subface_finite_element_values,
+                                                         cell,
+                                                         true,
+                                                         scratch.face_material_model_inputs);
+                    material_model->evaluate(scratch.face_material_model_inputs,
+                                             scratch.face_material_model_outputs);
+
+                    HeatingModel::HeatingModelOutputs face_heating_model_outputs(n_q_points, parameters.n_compositional_fields);
+                    heating_model_manager.evaluate(scratch.face_material_model_inputs,
+                                                   scratch.face_material_model_outputs,
+                                                   face_heating_model_outputs);
+
+                    //set up neighbor values
+                    scratch.neighbor_face_finite_element_values.reinit (neighbor_child, neighbor2);
+
+                    compute_material_model_input_values (current_linearization_point,
+                                                         scratch.neighbor_face_finite_element_values,
+                                                         neighbor_child,
+                                                         true,
+                                                         scratch.neighbor_face_material_model_inputs);
+                    material_model->evaluate(scratch.neighbor_face_material_model_inputs,
+                                             scratch.neighbor_face_material_model_outputs);
+
+                    HeatingModel::HeatingModelOutputs neighbor_face_heating_model_outputs(n_q_points, parameters.n_compositional_fields);
+                    heating_model_manager.evaluate(scratch.neighbor_face_material_model_inputs,
+                                                   scratch.neighbor_face_material_model_outputs,
+                                                   neighbor_face_heating_model_outputs);
+
+                    std::vector<types::global_dof_index> local_neighbor_dof_indices;
+                    local_neighbor_dof_indices.resize(scratch.face_finite_element_values.get_fe().dofs_per_cell);
```

Comment:
You can combine the two lines above by calling the constructor `std::vector<types::global_dof_index> local_neighbor_dof_indices (scratch.face_finite_element_values.get_fe().dofs_per_cell);`
Then everyone immediately knows how large this vector is going to be.